cpp/src/arrow/buffer-test.cc

```diff
@@ -52,6 +52,22 @@ TEST(TestBuffer, FromStdString) {
   ASSERT_EQ(static_cast<int64_t>(val.size()), buf.size());
 }
 
+TEST(TestBuffer, FromStdStringWithMemory) {
+  std::string* val = new std::string("hello, world");
```

Comment:
Use a stack allocated variable here: `std::string val = "hello, world";`. This will leak memory if one of the assertions fails.

Reply:
The reason I use "new" here is to check after the original string is gone the created buffer is still valid, which can verify if it actually owns its memory. So I need a way to deallocate the original string.

Reply:
Got it, I would say either leave a comment to that effect (so "new" haters won't undo the work later) or stack allocate in an ephemeral scope

```
{
  std::string whatever = "hello, world";
  RETURN_NOT_OK(Buffer::FromString(whatever, pool, &buf));
}
```